mlir/test/Dialect/Vector/vector-transfer-collapse-inner-most-dims.mlir

```diff
@@ -76,3 +76,24 @@ func.func @contiguous_inner_most_dim_out_of_bounds_2d(%arg0: memref<1x1xf32>) ->
 //  CHECK-NOT:   memref.subview
 //      CHECK:   %[[READ:.+]] = vector.transfer_read %[[SRC]]
 //      CHECK:   return %[[READ]] : vector<4x8xf32>
+
+// -----
+
+func.func @drop_inner_most_dim_for_transfer_write(%arg0: memref<1x512x16x1xf32, strided<[8192, 16, 1, 1], offset: ?>>, %arg1: vector<1x16x16x1xf32>, %arg2: index) {
```

Comment:
I'd add a few more tests:
* `memref` without strides - that would be a much simpler/shorter example,
* `memref` with unit strides (basically this test),
* `memref` with non-unit strides (effectively a negative test).

I might have another suggestion once I finish scanning the implementation.